From 060e4b124a8b1781504b8f894aa3c210cee513c1 Mon Sep 17 00:00:00 2001
From: Angus Hollands <goosey15@gmail.com>
Date: Tue, 15 Jun 2021 11:16:27 +0100
Subject: [PATCH] Bugfix: check for `file`-like objects in `from_parquet`

---
 src/awkward/operations/convert.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/awkward/operations/convert.py b/src/awkward/operations/convert.py
index 751ad727..dc20615e 100644
--- a/src/awkward/operations/convert.py
+++ b/src/awkward/operations/convert.py
@@ -3613,7 +3613,11 @@ def from_parquet(
             relative_to = source
             source = sorted(glob.glob(source + "/**/*.parquet", recursive=True))
 
-    if not isinstance(source, str) and isinstance(source, Iterable):
+    if (
+        not isinstance(source, str)
+        and isinstance(source, Iterable)
+        and not hasattr(source, "read")
+    ):
         source = [_regularize_path(x) for x in source]
         if relative_to is None:
             relative_to = os.path.commonpath(source)
-- 
2.32.0

